runner_test.go

```diff
@@ -0,0 +1,95 @@
+//go:build system
+
+package main
+
+import (
+	"os"
+	"testing"
+	"time"
+)
+
+func TestRunnerInitializeFinalize(t *testing.T) {
+	tb := func(n string) {
+		rn, err := newTestRunner()
+		testNoError(t, err)
+		rn.configuration.Backend = n
+		testNoError(t, rn.initialize())
+		testNoError(t, rn.finalize())
+	}
+
+	tb("ipset")
+	tb("nft")
+	tb("test")
+}
+
+func TestBackendInitializeInvalid(t *testing.T) {
+	tbi := func(n string) {
+		rn, err := newTestRunner()
+		testNoError(t, err)
+		rn.configuration.Backend = n
+		if err := rn.initialize(); err == nil {
+			t.Error("expected error")
+		}
+	}
+
+	tbi("")
+	tbi("unknownBackend")
+}
+
+func TestRunnerExecuteFlaky(t *testing.T) {
+	rn, err := newTestRunner()
+	testNoError(t, err)
+	testNoError(t, rn.initialize())
+	go rn.execute(false)
+	time.Sleep(100 * time.Millisecond)
+	rn.cancelChan <- true
+}
+
+func TestRunnerPerformActionFlaky(t *testing.T) {
+	pa := func(b string, a []string) {
+		rn, err := newTestRunner()
+		testNoError(t, err)
+		rn.configuration.Backend = b
+		rn.configuration.Rules["test"].Action = a
+		testNoError(t, rn.initialize())
+		go rn.execute(false)
+		time.Sleep(200 * time.Millisecond)
+		rn.cancelChan <- true
+	}
```

Comment:
We need to finalize the backend, or else the nft-Sets will be created multiple times.

Reply:
Is fixed